packages/next/server/render.tsx

```diff
@@ -281,22 +281,78 @@ function checkRedirectValues(
   }
 }
 
+function createRSCHook() {
+  const rscCache = new Map()
+  const decoder = new TextDecoder()
+
+  return (
+    writable: WritableStream,
+    id: string,
+    req: ReadableStream,
+    bootstrap: boolean
+  ) => {
+    let entry = rscCache.get(id)
+    if (!entry) {
+      const [renderStream, forwardStream] = req.tee()
+      entry = createFromReadableStream(renderStream)
+      rscCache.set(id, entry)
+
+      const transfomer = new TransformStream({
+        start(controller) {
+          if (bootstrap) {
+            controller.enqueue(
+              ReactDOMServer.renderToString(
```

Comment:
non-blocking, can do in later PR:
feel the enqueued work looks similar in `start` and `transform`, can be shared later